### DISCUSSION TITLE
Develop the Merge tool

### DISCUSSION THREAD

**Discussion by Pouria7** [dotnet/csharplang#6236] [General]
Hi...
It would be great if Microsoft created a Merge tool to upgrade some methods and syntaxes.
In this way, they can easily add or change some features like this:
https://github.com/dotnet/csharplang/discussions/6040

**Comment by HaloFour** [comment]
What does a Merge tool have to do with "upgrading" methods or syntaxes?  And how does that relate to the C# language specification?

**Reply by Pouria7** [reply]
https://doc.rust-lang.org/edition-guide/editions/transitioning-an-existing-project-to-a-new-edition.html
something like this:
https://doc.rust-lang.org/cargo/commands/cargo-fix.html

**Reply by HaloFour** [reply]
That's probably a better conversation for the [Roslyn](https://github.com/dotnet/roslyn) repository which deals with the compiler implementation and tooling.  Such tools would be independent from the language itself, even if they might enable the consideration of more breaking changes.

That said, I don't think that would change the team's position on breaking changes in general, especially when it comes to stylistic syntactic differences like the ones you suggested.  Even if there is a simple way within Visual Studio to convert a project from one language version to the other there would remain the problem of all of the existing resources (books, Internet, etc.) referencing different and incompatible dialects of the C# language.  IMO, it's not worth it except for the most extreme break-glass scenarios, and given the teams approach to NRTs I doubt much, if anything, would bubble up to that level.

**Comment by NGame1** [comment]
İ think he means replacing the new functions and methods with old ones like deprecated methods or classes with their replacements.
Currently it pops up as a warning and should be replace manually.
